portal/src/main/webapp/js/src/oncoprint/setup.js

```diff
@@ -2499,9 +2499,153 @@ window.CreateCBioPortalOncoprintWithToolbar = function (ctr_selector, toolbar_se
+			if (oncoprintData.CNA[currentGeneName] === undefined) {
+			    oncoprintData.CNA[currentGeneName] = {};
+			}
+			if (oncoprintData.MUTATIONS[currentGeneName] === undefined) {
+			    oncoprintData.MUTATIONS[currentGeneName] = {};
+			}
+			if (oncoprintData.MRNA[currentGeneName] === undefined) {
+			    oncoprintData.MRNA[currentGeneName] = {};
+			}
+			if (oncoprintData.PROTEIN[currentGeneName] === undefined) {
+			    oncoprintData.PROTEIN[currentGeneName] = {};
+			}
+			if (oncoprintData.FUSION[currentGeneName] === undefined) {
+			    oncoprintData.FUSION[currentGeneName] = {};
+			}
+			//Iterate over all patients/samples of the track and add them to oncoprintData
+			for (var s=0; s<currentTrackData.length; s++) {
+			    oncoprintData.CNA[currentGeneName][currentTrackData[s][patientOrSample]] = "";
+			    oncoprintData.MUTATIONS[currentGeneName][currentTrackData[s][patientOrSample]] = "";
+			    oncoprintData.MRNA[currentGeneName][currentTrackData[s][patientOrSample]] = "";
+			    oncoprintData.PROTEIN[currentGeneName][currentTrackData[s][patientOrSample]] = "";
+			    oncoprintData.FUSION[currentGeneName][currentTrackData[s][patientOrSample]] = "";
+			    if (currentTrackData[s].disp_cna !== undefined) {
+				oncoprintData.CNA[currentGeneName][currentTrackData[s][patientOrSample]] = cnaMap[currentTrackData[s].disp_cna] ? cnaMap[currentTrackData[s].disp_cna] : currentTrackData[s].disp_cna;
+			    }
+			    if (currentTrackData[s].disp_fusion !== undefined) {
+				oncoprintData.FUSION[currentGeneName][currentTrackData[s][patientOrSample]] = fusionMap[currentTrackData[s].disp_fusion] ? fusionMap[currentTrackData[s].disp_fusion] : currentTrackData[s].disp_fusion;
+			    }
+			    if (currentTrackData[s].disp_mrna !== undefined) {
+				oncoprintData.MRNA[currentGeneName][currentTrackData[s][patientOrSample]] = mrnaMap[currentTrackData[s].disp_mrna] ? mrnaMap[currentTrackData[s].disp_mrna] : currentTrackData[s].disp_mrna;
+			    }
+			    if (currentTrackData[s].disp_prot !== undefined) {
+				oncoprintData.PROTEIN[currentGeneName][currentTrackData[s][patientOrSample]] = proteinMap[currentTrackData[s].disp_prot] ? proteinMap[currentTrackData[s].disp_prot] : currentTrackData[s].disp_prot;
+			    }
+			    if (currentTrackData[s].disp_mut !== undefined) {
+				oncoprintData.MUTATIONS[currentGeneName][currentTrackData[s][patientOrSample]] = mutationMap[currentTrackData[s].disp_mut] ? mutationMap[currentTrackData[s].disp_mut] : currentTrackData[s].disp_mut;
+			    }
+			}
+		    }
+		    
+		    //Put all the information of the oncoprintData in a variable with tabular form
+		    var content = 'track_name\ttrack_type';
+		    //Add the cases to the content
+		    for (var i=0; i<caseNames.length; i++) {
+			content += '\t'+caseNames[i];
+		    }
+		    //Add final header line
+		    content += '\n';
+		    
+		    //Iterate over oncoprintData and write it to content
+		    for (var j in oncoprintData) {
```

Comment:
I dont think this type of iteration is stable with respect to order, which would be a problem when generating the table?